storage/src/main/java/org/commonjava/storage/pathmapped/core/PathDBOutputStream.java

```diff
@@ -74,16 +90,21 @@
     }
 
     @Override
-    public void write( int b ) throws IOException
+    public void write ( byte[] b, int off, int len )
+            throws IOException
     {
+        if (( off | len | ( b.length - ( len + off )) | ( off + len )) < 0 )
+            throw new IndexOutOfBoundsException();
+
         try
         {
-            super.write( b );
-            size += 1;
-            byte by = (byte) ( b & 0xff );
+            out.write( b, off, len );
+            size += len;
             if ( checksumCalculator != null )
             {
-                checksumCalculator.update( by );
+                boolean isSubArray = 0 != off || len < b.length;
+                byte[] sub = ( isSubArray ? Arrays.copyOfRange( b, off, len ) : b );
+                checksumCalculator.update( sub );
```

Comment:
As commented in galley pr, let's leverage the digester's method, avoid doing the sub-array by ourself.

Reply:
Yes let's add a method here as well.